tracee-ebpf/tracee/tracee.go

```diff
@@ -1007,23 +1007,26 @@ func boolToUInt32(b bool) uint32 {
 	return uint32(0)
 }
 
-func getFileHash(fileName string) string {
-	f, _ := os.Open(fileName)
-	if f != nil {
-		defer f.Close()
-		h := sha256.New()
-		_, _ = io.Copy(h, f)
-		return hex.EncodeToString(h.Sum(nil))
+func computeFileHash(fileName string) (string, error) {
+	f, err := os.Open(fileName)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	h := sha256.New()
+	_, err = io.Copy(h, f)
+	if err != nil {
+		return "", err
 	}
-	return ""
+	return hex.EncodeToString(h.Sum(nil)), nil
 }
 
 func (t *Tracee) updateFileSHA() {
 	for k, v := range t.profiledFiles {
 		s := strings.Split(k, ".")
 		exeName := strings.Split(s[1], ":")[0]
 		filePath := fmt.Sprintf("%s.%d.%s", s[0], v.FirstExecutionTs, exeName)
-		fileSHA := getFileHash(filePath)
+		fileSHA, _ := computeFileHash(filePath)
```

Comment:
Is it intentional to skip error handling here? If not we may miss some logical errors. Shouldn't updateFileSHA() return error?

Reply:
So yeah I agree with you. I left it without error check as it was before since I am not sure how we should handle error in this case. Maybe `updateFileSHA` should return error as well?

Reply:
I'm always following this "fail fast" or "immediate and visible failure" principle. Unless there are good reasons for that. Here we should probably propagate this error to indicate abnormal behaviour of tracee. Normally computing hashes should just work

Reply:
right, so this function should be changed as well - updateFileSha should return an error as well.